### PR TITLE
Benny/single nft endpoint

### PR DESCRIPTION
**Pull request by bennycio**
Changes:

- **Changed** many files' formatting with go fmt ./... (I apologize this changed so many files, it happens when I save in vscode)
- **Added** endpoint for retrieving nft by id
- **Added** function for retrieving nft by id from mongo
- **Added** test function for creating an nft then retrieving that nft and finally deleting the nft by id from mongo

Files to review that were not just adjusted by go fmt:
- glry_db/t__nft_test.go
- glry_db/glry_nft.go
- glry_lib/glry_http_handlers.go

### CODE REVIEW

**Review by fbrcsftnr — APPROVED** [review]

File: glry_lib/glry_http_handlers.go [hunk 1]
```diff
@@ -137,20 +136,60 @@ func HandlersInit(pRuntime *glry_core.Runtime) {
...
+
+				nftIDstr := qMap.Get("id")
+
+				if nftIDstr == "" {
+					// is this the right way to create an error for gf_core?
+					return nil, gf_core.Error__create("nft id not found in query values",
+						"http_client_req_error",
+						map[string]interface{}{
+							"uri": "/glry/v1/nfts/get",
+						}, nil, "glry_core", pRuntime.RuntimeSys)
```
Comment (kaitoo1) [inline glry_lib/glry_http_handlers.go:171]:
```suggestion
						}, nil, "glry_lib", pRuntime.RuntimeSys)
```

Comment (kaitoo1) [reply glry_lib/glry_http_handlers.go:171]: super small comment but I think `glry_core` should be `glry_lib` since this file lives in `glry_lib`

**Review by kaitoo1 — APPROVED** [review]

**Comment by kaitoo1** [comment]
Nice work!! that was fast  🔥

Looks great overall - I have one comment about the NOT FOUND case, and left a smaller nit comment above.

I see that for the NOT FOUND case where the requested nft ID doesn't exist, we return `{"data":{"nfts":[]},"status":"OK"}`
Could we return an error instead of status OK and an empty array of nfts? Similar to how you return an error if no id param is provided?

That would make it easier to handle NOT FOUND cases on the client, and also makes it easier to classify as an error case for logs and other uses

File: glry_lib/glry_http_handlers.go (hunk 1 above)
Comment (fbrcsftnr) [reply glry_lib/glry_http_handlers.go:171]: awesome catch

**Comment by fbrcsftnr** [comment]
@kaitoo1 agreed that we should return an error in this scenario

**Comment by fbrcsftnr** [comment]
example error creation: https://github.com/gallery-so/go-gallery/blob/f50e1b3b846dcac6a6be74203d80c64b212ad6e1/glry_lib/glry_auth_jwt.go#L104

i believe you just need to return this error out of the top-level HTTP handler and it'll send down a 200 with an error message attached

File: glry_lib/glry_http_handlers.go (hunk 1 above)
Comment (bennycio) [reply glry_lib/glry_http_handlers.go:171]: Thank you! Didn't even notice!

**Comment by bennycio** [comment]
Thank you guys! Will handle the not found case as soon as I can with a similar error as the id not found error.

**Pull request merged**